graylog2-server/src/main/java/org/graylog2/indexer/cluster/Cluster.java

```diff
@@ -130,28 +136,48 @@ public JsonArray catNodes(String fields) {
     }
 
     public String nodeIdToName(String nodeId) {
-        final JsonObject nodeInfo = getNodeInfo(nodeId);
-        return nodeInfo == null ? "UNKNOWN" : nodeInfo.getAsJsonPrimitive("name").getAsString();
-
+        return getNodeInfo(nodeId)
+                .map(nodeInfo -> nodeInfo.get("name"))
+                .filter(JsonElement::isJsonPrimitive)
+                .map(JsonElement::getAsJsonPrimitive)
+                .filter(JsonPrimitive::isString)
+                .map(JsonPrimitive::getAsString)
+                .orElse("UNKNOWN");
```

Comment:
Can we turn the result type into an `Optional<String>` instead? Rendering it as unknown would be up to the presentation layer then.